lib/eslint/eslint.js

```diff
@@ -314,7 +357,46 @@ async function loadFlatConfigFile(filePath) {
         delete require.cache[filePath];
     }
 
-    const config = (await import(fileURL)).default;
+    const isTS = isFileTS(filePath) && hasUnstableTSConfigFlag;
+
+    const isBun = isRunningInBun();
+
+    const isDeno = isRunningInDeno();
+
+    if (isTS && !isDeno && !isBun) {
+
+        /**
+         * @type {typeof import("jiti").createJiti}
+         */
+        const createJiti = await import("jiti").then(jitiModule => {
+
+            if (typeof jitiModule?.createJiti === "function") {
+                return jitiModule.createJiti;
+            }
+
+            process.emitWarning("You are using an outdated version of the 'jiti' library. The top-level await syntax is not supported in this version. Please update to the latest version of 'jiti' to ensure compatibility and access to the latest features.", "ESLintOutdatedJitiWarning");
+
+            return jitiModule.default;
```

Comment:
Usually we don't allow a range of multiple major versions for dependencies or (devDependencies), so I'm not sure if we should allow different versions of jiti with a warning here. If we do, we should probably run the relevant unit tests with multiple versions of jiti as well (currently, some code paths are uncovered and the TLA tests will fail with jiti v1). I think that for a start, it would be sufficient to support only jiti v1 or only v2. Opinions?

Reply:
I agree. Let's just keep this simple to start and go with jiti v1 since v2 isn't ready yet. This feature is experimental so we can always change it before its finalized. We can note in the docs that top-level await is not yet supported and leave it at that.

Reply:
I was hoping that we could use jiti v2 stable, but it's not ready yet and we don't know how long it will take. If we want to switch this PR to jiti v1, I think these are the changes needed:

1. Remove the top-level `await` tests (we can re-add them later)
1. Re-apply the fix to [tests/lib/cli-engine/lint-result-cache.js](https://github.com/eslint/eslint/pull/18134/files/f801ddccd82a3c4636d32d6fede0bc78a0e03075#diff-9ce1c808abb04f9070970d1a3207987e0a09524b759b287b731e4775a7ca91eb) so it doesn't overwrite `process.version`
1. In package.json, change the jiti version in `devDependencies`
1. Remove the jiti v2 specific logic in `loadFlatConfigFile()`

Reply:
I suggest using jiti v2 constraint for the sake of performance and better forward ecosystem compatibility. The reason I still kept it under beta is that we are testing v2 against multiple ecosystems and projects to make sure it has no edge cases, and once released, we are going to push it ever.

Reply:
> Usually we don't allow a range of multiple major versions for dependencies

It seems we have to do that here - to avoid breaking changes, it's written `"peerDependencies": {"jiti": "*"}`.